atc/lidar/scanner_test.go

```diff
@@ -168,6 +168,23 @@ var _ = Describe("Scanner", func() {
 						Expect(manuallyTriggered).To(BeFalse())
 					})
 				})
+
+				Context("when a put-only resource has a parent-type", func() {
+					BeforeEach(func() {
+						By("checkFactory.Resources should not return any put-only resources")
+						fakeResourceType.NameReturns("put-only-custom-type")
+						fakeResourceType.PipelineIDReturns(1)
+					})
+
+					It("creates a check for only the parent and not the put-only resource", func() {
+						Expect(fakeCheckFactory.TryCreateCheckCallCount()).To(Equal(2))
+
+						By("first check is for the default resource")
+						_, checkable, _, _, manuallyTriggered := fakeCheckFactory.TryCreateCheckArgsForCall(1)
```

Comment:
This test is failing https://ci.concourse-ci.org/builds/142874260#L6087a117:1177, should this be:
```suggestion
						_, checkable, _, _, manuallyTriggered := fakeCheckFactory.TryCreateCheckArgsForCall(0)
```